src/diagnostics.rs

```diff
@@ -157,6 +157,46 @@ pub fn report_error<'tcx, 'mir>(
         }
     };
 
+    let mut stacktrace = ecx.generate_stacktrace();
+    let has_local_frame = stacktrace.iter().any(|frame| frame.instance.def_id().is_local());
```

Comment:
Can you move this `let` into the `Short` match arm?

Reply:
No, I factored this out so that it can be used below as well to skip printing the note about details were omitted if no details were omitted.